tstest/integration/integration_test.go

```diff
@@ -521,6 +522,164 @@ func TestIncrementalMapUpdatePeersRemoved(t *testing.T) {
 	d2.MustCleanShutdown(t)
 }
 
+func TestCapMapPacketFilter(t *testing.T) {
+	tstest.Shard(t)
+	tstest.Parallel(t)
+	env := newTestEnv(t)
+
+	n1 := newTestNode(t, env)
+	d1 := n1.StartDaemon()
+	n1.AwaitListening()
+	n1.MustUp()
+	n1.AwaitRunning()
+
+	all := env.Control.AllNodes()
+	if len(all) != 1 {
+		t.Fatalf("expected 1 node, got %d nodes", len(all))
+	}
+	tnode1 := all[0]
+
+	n2 := newTestNode(t, env)
+	d2 := n2.StartDaemon()
+	n2.AwaitListening()
+	n2.MustUp()
+	n2.AwaitRunning()
+
+	all = env.Control.AllNodes()
+	if len(all) != 2 {
+		t.Fatalf("expected 2 node, got %d nodes", len(all))
+	}
+	var tnode2 *tailcfg.Node
+	for _, n := range all {
+		if n.ID != tnode1.ID {
+			tnode2 = n
+		}
+	}
+	if tnode2 == nil {
+		t.Fatalf("failed to find second node ID (two dups?)")
+	}
+
+	t.Logf("node1=%v, node2=%v", tnode1.ID, tnode2.ID)
+
+	n1.AwaitStatus(func(st *ipnstate.Status) error {
+		if len(st.Peer) != 1 {
+			return fmt.Errorf("got %d peers; want 1", len(st.Peer))
+		}
+		peer := st.Peer[st.Peers()[0]]
+		if peer.ID == st.Self.ID {
+			return errors.New("peer is self")
+		}
+		return nil
+	})
+
+	// Check that n2 is reachable from n1 with the default packet filter.
+	if err := n1.PingICMP(n2); err != nil {
+		t.Fatalf("ping: %v", err)
+	}
+
+	t.Logf("setting packet filter with a cap")
+	if !env.Control.AddRawMapResponse(tnode2.Key, &tailcfg.MapResponse{
+		PacketFilter: []tailcfg.FilterRule{
+			{
+				SrcIPs: []string{"cap:foobar"},
+				DstPorts: []tailcfg.NetPortRange{{
+					IP:    "*",
+					Ports: tailcfg.PortRange{First: 0, Last: 65535},
+				}},
+			},
+		},
+	}) {
+		t.Fatalf("failed to add map response")
+	}
+
+	// Wait until n2 is no longer reachable from n1.
+	if err := tstest.WaitFor(5*time.Second, func() error {
+		if err := n1.PingICMP(n2); err == nil {
+			t.Fatal("ping successful, wanted an error")
```

Comment:
Are we supposed to return an error to initiate a retry?

```suggestion
			return errors.New("ping successful, wanted an error")
```